### PR TITLE
HPCC-18954 Update eclcc options in docs

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>
@ghalliday Please review 

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml [hunk 1]
```diff
@@ -387,6 +377,49 @@
...
+                <entry>Use specified password in DFS filename resolution
+                (blank to prompt)</entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </informaltable></para>
+
+      <?hard-pagebreak ?>
+
       <para>Other options:</para>
```
Comment (ghalliday) [inline docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml]: Worth indicating this list can be obtained with eclcc -help -v

**Comment by ghalliday** [comment]
-platform=hthor currently described differently from -platform=roxie, it would make sense if it was the same.
Also, roxie is now the default for a stand alone executable.

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml [hunk 2]
```diff
@@ -398,16 +431,22 @@
...
-                Python, R, etc.)</para><para><emphasis
-                role="bold">pipe</emphasis>: Allow the PIPE command to send
-                data to an external program.</para><para><emphasis
-                role="bold">all</emphasis>: Allow all features</para></entry>
+                all)<para><emphasis role="bold">cpp</emphasis>: Allow embedded
+                code within ECL (e.g., c++, JAVA, Javascript, Python, R,
+                etc.)</para><para><emphasis role="bold">pipe</emphasis>: Allow
+                the PIPE command to send data to an external
+                program.</para><para><emphasis role="bold">all</emphasis>:
+                Allow all features</para></entry>
```
Comment (ghalliday) [inline docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml]: userECL: Allow code that is not found via the ecl include paths
datafile: Access to datafiles from ecl.
extern: access to an external service function

There is also a --allowsigned option which only allows access to the feature from signed code.

**Review by ghalliday — COMMENTED** [review]
A few extra items worth fixing at the same time.

**Comment by ghalliday** [comment]
@JamesDeFabia a couple of comments

**Comment by JamesDeFabia** [comment]
@ghalliday Edits made as requested, back to you

**Comment by AttilaVamos** [comment]
Those Roxie errors are come from the master branch and not related to this PR.

**Comment by ghalliday** [comment]
@JamesDeFabia please can you squash the commits.

**Comment by JamesDeFabia** [comment]
@ghalliday squashed as requested

**Pull request merged**